go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/netobserv/netobserv-ebpf-agent
 
-go 1.19
+go 1.21
+
+toolchain go1.21.1
```

Comment:
why u need this ? it seems to fail lint check in ur PR pls run
`make lint` on ur view

Reply:
I didn't realize this but the behavior changed in go 1.21 so that it's a compile error if a dependency has a `go` directive with a higher `go` version than what you're compiling it with:
From https://go.dev/doc/modules/gomod-ref#go:
> The go directive sets the minimum version of Go required to use this module. Before Go 1.21, the directive was advisory only; now it is a mandatory requirement: Go toolchains refuse to use modules declaring newer Go versions.

So, I think things will fail to build on go 1.21 until the `go.mod` is updated like I show here, but I did switch back to go 1.19 and was able to run `go mod tidy` and `go mod vendor` and run `make build`, so I removed this change.